### DISCUSSION TITLE
GenerateReleaseNotes (Xplat version v3) Question about artifact builds

### DISCUSSION THREAD

**Discussion by rsuela** [rfennell/AzurePipelines#995] [Q&A]
Hi,

I have a _classic release pipeline_ with Artifacts containing repositories, and build pipelines. The GenerateReleaseNotes works well and I am able to create release notes with versions of the artifact build pipelines and repositories (including WIs). 

Now, I am looking to move the release notes task to a **multi-stage yaml pipeline**. I've added the artifacts under resources in pipeline, the yaml build deployment is working correctly except for the release notes. I cannot get the details of the artifact builds and their work items.

```resources:
  repositories: 
  - repository: 'CosmosBusinessData'
    type: git
    name: 'CosmosBusinessData'
    ref: 'refs/heads/master'

  - repository: 'UIComponent'
    type: git
    name: 'UIComponent'
    ref: 'refs/heads/master' 
    
  pipelines:
  - pipeline: APIBuilds
    source: APIBuilds

  - stage: 'Generate Release Notes'
    jobs:
    - job: 
      steps:
      - checkout: 'CosmosBusinessData'
      - checkout: 'UIComponent'
      - download: 'APIBuilds'
      - task: XplatGenerateReleaseNotes@3
        ....
```

What I want to do is get the information of the repositories, pipeline builds and their WIs then store it in the release notes. After that, publish it in Wiki where the team can find out which version is deployed in an environment.

Is this currently possible for multi-stage yaml pipeline? 

Appreciate any info. Cheers!

**Comment by rfennell** [comment]
There no sure functionality to allow a multi-stage YAML build to inspect the artefact versions in either my Legacy (PS) or Current (Node) release notes task.

I am not adding any new features to the PS version, the architecture makes changes too much work, but would consider adding this to the Node version where new features are much easier to implement.

**Reply by rsuela** [reply]
Cool. Thank you.

**Reply by rfennell** [reply]
I did a bit of digging and it appears there is no way via the API to get the consumed artefacts at present (see [this post](https://stackoverflow.com/questions/64572921/get-consumed-pipeline-artifacts-from-azure-devops-build-via-rest-api)). 

So I unless there is an undocumented API call, I doubt  will be adding this feature anytime soon

**Comment by rsuela** [comment]
Thanks Richard. I understand that there is a limitation, I hope this gets added in the future. 

As for my work, I plan to add a generate release notes task on each build and save it to a file then publish in the build. Then in the multi stage pipeline, I will download the file and append in on the final release notes. Hope it works.  :)

**Comment by rfennell** [comment]
With some help on the API have managed to get the consumed items. It will be released as 3.50.x and the Handlebars entry will be something like

```
                  ## Global list of ConsumedArtifacts ({{consumedArtifacts.length}})
                  {{#forEach consumedArtifacts}}
                  * {{this.artifactCategory}} -  {{this.versionName}}
                  {{/forEach}}
```

**Reply by rsuela** [reply]
Cheers! Thank you for this.
